packages/hooks/src/data/QueryData.ts

```diff
@@ -118,6 +115,85 @@ export class QueryData<TData, TVariables> extends OperationData {
     delete this.previousData.result;
   }
 
+  public getOptions() {
+    const options = super.getOptions();
+    const lazyOptions = this.lazyOptions || {};
+    const updatedOptions = {
+      ...options,
+      variables: {
+        ...options.variables,
+        ...lazyOptions.variables
+      },
+      context: {
+        ...options.context,
+        ...lazyOptions.context
+      }
+    };
+
+    // Triggering a query in lazy mode overrides `skip` settings.
+    if (options.lazy) {
+      updatedOptions.skip = false;
+    }
+
+    return updatedOptions;
+  }
+
+  private initialLazyResult(): QueryTuple<TData, TVariables> {
+    return [
+      {
+        loading: false,
+        networkStatus: NetworkStatus.ready,
+        called: false,
+        data: undefined
+      } as QueryResult<TData, TVariables>,
+      this.executeLazy.bind(this)
+    ];
+  }
+
+  private executeLazy(options?: QueryLazyOptions<TVariables>) {
```

Comment:
```suggestion
  private executeLazy = (options?: QueryLazyOptions<TVariables>) => {
```
Let’s use class property syntax here to bind `this.executeLazy` once, in the constructor.